tests/DriverManagerTest.php

```diff
@@ -173,15 +177,22 @@ public function testDatabaseUrlDeprecated($url, $expected): void
     }
 
     /**
-     * @param mixed $url
-     * @param mixed $expected
+     * @param array<string, mixed>|string $url
+     * @param array<string, mixed>|false  $expected
      *
      * @dataProvider databaseUrls
      */
     public function testDatabaseUrl($url, $expected): void
     {
         if (is_array($url)) {
+            if (isset($url['driverClass'])) {
+                self::markTestSkipped(
+                    'Legacy test case: Merging driverClass into the parsed parameters has to be done in userland now.',
+                );
+            }
+
             ['url' => $url] = $options = $url;
+            unset($options['url']);
```

Comment:
If I don't unset the URL here, it's actually parsed twice which is why the `driverClass` tests still worked although they shouldn't have. 😓